jdaviz/configs/default/plugins/line_lists/line_lists.py

```diff
@@ -88,29 +106,173 @@ def _on_viewer_data_changed(self, msg=None):
+        else:
+            raise NotImplementedError(f"RedshiftMessage with param {param} not implemented.")
 
-        self._units["x"] = str(self._viewer_spectrum.spectral_axis.unit)
-        self._units["y"] = str(self._viewer_spectrum.flux.unit)
+    def _velocity_to_redshift(self, velocity):
+        """
+        Convert a velocity to a relativistic redshift.
+        """
+        beta = velocity / c
+        return np.sqrt((1 + beta) / (1 - beta)) - 1
 
-        self._bounds["min"] = self._viewer_spectrum.spectral_axis[0]
-        self._bounds["max"] = self._viewer_spectrum.spectral_axis[-1]
+    def _redshift_to_velocity(self, redshift):
+        """
+        Convert a relativistic redshift to a velocity.
+        """
+        zponesq = (1 + redshift) ** 2
+        return c * (zponesq - 1) / (zponesq + 1)
+
+    @observe('rs_slider')
+    def _on_slider_updated(self, event):
+        if self._rs_disable_observe:
+            return
+
+        # NOTE: _on_rs_redshift_updated will handle updating rs_rv
+        # NOTE: rs_redshift could be a string if entered by the user, so we cast old value to float
+        self.rs_redshift = float(self.rs_redshift) + event['new'] - event['old']
+
+    @observe('rs_redshift')
+    def _on_rs_redshift_updated(self, event):
+        if not self._rs_disable_observe:
+            if isinstance(event['new'], str) and not len(event['new']):
+                # empty string, we don't want to revert yet because then
+                # the user can never delete the entry and type something new
+                # so we'll just leave empty
+                return
+            value = float(event['new'])
+            self._rs_disable_observe = True
+            self.rs_rv = self._redshift_to_velocity(value).to('km/s').value
+            self._rs_disable_observe = False
+
+        # update all lines, self._global_redshift, and emit message back to Specviz helper
+        z = u.Quantity(self.rs_redshift)
+        line_list = self.app.get_viewer('spectrum-viewer').spectral_lines
+        if line_list is not None:
+            line_list["redshift"] = z
+            # Replot with the new redshift
+            line_list = self.app.get_viewer('spectrum-viewer').plot_spectral_lines()
+
+        # Send the redshift back to the Specviz helper (and also trigger
+        # self._update_global_redshift)
+        msg = RedshiftMessage("redshift", z.value, sender=self)
+        self.app.hub.broadcast(msg)
+
+    @observe('rs_rv')
+    def _on_rs_rv_updated(self, event):
+        if self._rs_disable_observe:
+            return
+
+        if isinstance(event['new'], str) and not len(event['new']):
+            # empty string, we don't want to revert yet because then
+            # the user can never delete the entry and type something new
+            # so we'll just leave empty
+            return
+
+        value = float(event['new'])
+        redshift = self._velocity_to_redshift(value*u.km/u.s).value
```

Comment:
Build the composite unit first.
```suggestion
        redshift = self._velocity_to_redshift(value * (u.km/u.s)).value
```